src/linker/Linker.Steps/MarkStep.cs

```diff
@@ -456,6 +474,32 @@ void ProcessMarkedTypesWithInterfaces ()
 			}
 		}
 
+		void ProcessDynamicCastableImplementationInterfaces ()
+		{
+			// We may mark an interface type later on.  Which means we need to reprocess any time with one or more interface implementations that have not been marked
+			// and if an interface type is found to be marked and implementation is not marked, then we need to mark that implementation
+
+			for (int i = 0; i < _dynamicInterfaceCastableImplementationTypes.Count; i++) {
+				var type = _dynamicInterfaceCastableImplementationTypes[i];
+
+				Debug.Assert (TypeHasDynamicInterfaceCastableImplementationAttribute (type));
+
+				// If the type has already been marked, we can remove it from this list.
+				if (Annotations.IsMarked (type)) {
+					_dynamicInterfaceCastableImplementationTypes.RemoveAt (i--);
+					continue;
+				}
+
+				foreach (var iface in type.Interfaces) {
+					if (Annotations.IsMarked (iface.InterfaceType)) {
+						MarkType (type, new DependencyInfo (DependencyKind.DynamicInterfaceCastableImplementation, iface.InterfaceType), type);
```

Comment:
What is the intention here, is marking type definition only enough? It would be nice to add a comment about the reason.

Reply:
Yes, marking the type definition only is enough. Because the linker will automatically mark interface implementations of marked interfaces on a marked type and mark implementations of methods from those interfaces, the linker already does the rest of the linking we need to do once we mark the type definition.

I'll add a comment.